Adding new modifiers

alexjiao125 opened the pull request:
added frequency, fade, and counter modules

alexjiao125 committed:
added counter module

alexjiao125 committed:
fixed fade module

alexjiao125 committed:
Added frequency module

myjeeed commented on Modules/Modifier/Counter/ArduinoCodes/CounterATtiny/CounterATtiny.ino:
```diff
@@ -24,9 +24,15 @@
 
 #include <SignalProcessing.h>
 
+#define THRESH_LOW 300
```
@alexjiao125  can you add a small comment about these two variables?

myjeeed commented on Modules/Modifier/Counter/ArduinoCodes/CounterATtiny/CounterATtiny.ino:
```diff
@@ -24,9 +24,15 @@
 int input_pin = 3;                           //Pin 2 on ATtiny
 int output_pin = 1;                          //Pin 6 on ATtiny
 int filter_size = 15;                        //Noise reduction filter size
+short countable = 1; 
```
@alexjiao125  what is countable?

alexjiao125 replied:
If countable is 1, the counter can be incremented; it is used to determine whether or not a LOW value has been read since the last HIGH value.  If not, the counter is not increased.

myjeeed commented on Modules/Modifier/Fade/ArduinoCodes/FadeATtiny/FadeATtiny.ino:
```diff
@@ -65,8 +65,8 @@ void loop()
-  
-if(on_trigger == 1 && fading == 0)
+    
+  if(on_trigger == 1 && fading == 0)
```
@alexjiao125  can you also add an if statement for `if(on_trigger == 1 && fading == 1)`? we could add up some value to the brightness upon receiving a new pulse while it is still in a fading state.
Also, how is the fade happening? I do not see something such as `brightness--;` or any other variable being decreased over time.

myjeeed commented on Modules/Sensor/PulseSensor/EagleDesigns/.PulseSensor.brd.lck:
```diff
@@ -0,0 +1,3 @@
+alexj_000
```
delete these two files from this branch.

alexjiao125 committed:
added comments

alexjiao125 committed:
added more comments

alexjiao125 committed:
Tweaked modifiers

Changed fade so it only does it on HIGH pulses that come after LOW
pulses.  Finalized PulseMeter and made an ATtiny version of code.

myjeeed commented on Modules/Modifier/Fade/ArduinoCodes/FadeATtiny/FadeATtiny.ino:
```diff
@@ -47,69 +52,47 @@ void setup()
+  int pot_value;
+  int input_val = cutAndMap(input.filteredAnalogRead(AVERAGE), 50, 975, 4, 1023);
+
+  if(input_val > THRESH_HIGH){
```
check for `THRESH_LOW` in your if-else statements as well, and then we can merge this branch.

alexjiao125 committed:
Corrected the Fade Module

Did not work with the button at first.  Also was not correctly using the
LOW threshold.

alexjiao125 committed:
Changed filter sizes

Changed filter sizes from 15 to 5 on Counter and PulseMeter modules.

alexjiao125 committed:
changed fade module

allows for changing fade time mid-fade now
